src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs

```diff
@@ -152,14 +155,29 @@ internal abstract partial class AbstractNavigateToSearchService
         }
 
         private static async Task<ImmutableArray<SearchResult>> ComputeSearchResultsAsync(
-            Project project, Document searchDocument,
+            Project project, ImmutableArray<Document> priorityDocuments, Document searchDocument, 
             PatternMatcher nameMatcher, PatternMatcher containerMatcherOpt,
             DeclaredSymbolInfoKindSet kinds,
             ArrayBuilder<PatternMatch> nameMatches, ArrayBuilder<PatternMatch> containerMatches,
             CancellationToken cancellationToken)
         {
             var result = ArrayBuilder<SearchResult>.GetInstance();
-            foreach (var document in project.Documents)
+
+            // Prioritize the active documents if we have any.
+            var highPriDocs = priorityDocuments.Where(d => project.ContainsDocument(d.Id))
```

Comment:
so, this does filtering to find docs that belong to current project. then is there a reason the higher up, caller needs to filter out and create new array for each project?

Reply:
sorry, i don't understand hte question.  :)    

>  then is there a reason the higher up, caller needs to filter out and create new array for each project?

What are you referring to?  The caller's job is to say which documents it would prefer be searched first.  This code then uses that to order things properly to respect the caller's wishes.

Reply:
so you do this per SearchAsync - priorityDocs.ToImmutableArray() - for a project.

I am asking if you create var list = activeDoc,Cocat(restofVisibleDoc) and call all SearchAsync(project, list)

will result or order of documents processed change?

otherwise, we bother to create list again and again?

Reply:
This would have different semantics from what i want.  The semantics i want are:

1. process *only* the project with the active document *first*.  Do not do anything else while processing this project.
2. process hte projects for the visible documents in *parallel*.  
3. process anything remaining in parallel.

Placing hte active document at the head of the visible-doc list, and then processing that list in parallel would have a different experience.  Specifically, the most relevant project would now have to compete for CPU time against all the projects that have visible docs.

Reply:
I dont think any behavior you have currently will change even if you passed in a priority list that contains active document at the head and all other visible document after that. rather than creating new list for each project to filter out documents that belong to one project.

Reply:
I'm waiting for this thread to be resolved, in order to merge the PR. Thanks

Reply:
> I dont think any behavior you have currently will change even if you passed in a priority list 

... why not?  I really don't understand your concern here.  Can you please explain it fully?  If necessary, can you please come to gitter.im so we can discuss this?  I'm really not understanding why you think this would not work.

> that contains active document at the head and all other visible document after that.

As i've explained, we process all visible documents in parallel.  So if the active document is at the front, that means that we'll process it in parallel with the visible documents.  That's *not* what i want.  I want the active document to be processed entirely independently, so it's not sharing any CPU with any other processing.

Reply:
That makes sense to me. 
@heejaechang Are you satisfied too?